### ISSUE TITLE
ntpd: compilation fails due to missing libevent_pthread dependency

### ISSUE THREAD

**Issue by KloudJack** [openwrt/packages#10307]
Maintainer: @tripolar 
Environment: OpenWRT snapshot r11266-34939711a5 - packages @ fa790f0

Description:
Building master of openwrt and all feeds with any ntpd package selected fails due to missing libevent_pthreads dependency during linking.

I'm building with libstdc++ support instead of uClibc/++.
This could make a difference if uClibc pthreads support evaluates differently in ntpd's automake definitions vs. those of GNU's.

```
Version <sntp 4.2.8p13@1.3847-o Sun Oct 20 23:33:27 UTC 2019 (1)>
env CCACHE_DISABLE=1 arm-openwrt-linux-muslgnueabi-gcc -DHAVE_CONFIG_H -I.  -I./../include -I./../lib/isc/include -I./../lib/isc/pthreads/include -I./../lib/  isc/unix/include -I./libopts  -I/home/kloudjack/projects/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/usr/include -D_FORTIFY_SOURCE=2 -I/    home/kloudjack/projects/openwrt/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/usr/include -I/home/kloudjack/projects/openwrt/staging_dir/target-      arm_cortex-a15+neon-vfpv4_musl_eabi/include -I/home/kloudjack/projects/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.3.0_musl_eabi/usr/        include -I/home/kloudjack/projects/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.3.0_musl_eabi/include/fortify -I/home/kloudjack/projects/     openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.3.0_musl_eabi/include  -Wall -Wcast-align -Wcast-qual -Wmissing-prototypes -Wpointer-arith -     Wshadow -Winit-self -Wstrict-overflow   -Wno-strict-prototypes -fPIE -fPIC -fstack-protector-all -O1 -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -Wno- error=unused-but-set-variable -Wno-error=unused-result -mfloat-abi=hard -ffile-prefix-map=/home/kloudjack/projects/openwrt/build_dir/target-arm_cortex-a15+    neon-vfpv4_musl_eabi/ntp-4.2.8p13=ntp-4.2.8p13 -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro  -c version.c -o version.o
<command-line>: warning: "_FORTIFY_SOURCE" redefined
<command-line>: note: this is the location of the previous definition
  CCLD     sntp
/home/kloudjack/projects/openwrt/staging_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.3.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/8.3.0/../../../../  arm-openwrt-linux-muslgnueabi/bin/ld: cannot find -levent_pthreads
collect2: error: ld returned 1 exit status
Makefile:887: recipe for target 'sntp' failed
```

**Comment by karlp** [comment]
@tripolar  This is also broken in 19.07 branch

**Comment by karlp** [comment]
SNTP is the only user of libevent, are you sure this is the right fix?

**Comment by karlp** [comment]
and by that, I meant the sntp binary, not sntp support in the toolset itself

**Comment by KloudJack** [comment]
@karlp libevent was previously defined as a dependency for ntp-keygen when I created the PR to fix this.

Checking the ntp 4.2.8p13 Makefile and running a build without sntp, I agree with you that libevent is only required for the sntp binary.

I've created a new PR which includes the sntp binary as a part of the ntp-utils package while removing it and its libevent dependencies from the other packages.
